src/Servers/Connections.Abstractions/src/BaseConnectionContext.cs

```diff
@@ -45,6 +46,11 @@ public abstract class BaseConnectionContext : IAsyncDisposable
         /// </summary>
         public virtual EndPoint? RemoteEndPoint { get; set; }
 
+        /// <summary>
+        /// Gets or sets the socket for this connection.
+        /// </summary>
+        public virtual Socket? Socket { get; set; }
```

Comment:
This shouldn't be on the base, it should be defined in the sockets transport. It should probably also be get only.